src/main/java/org/terasology/launcher/packages/PackageDatabase.java

```diff
@@ -43,39 +46,40 @@
 
     private static final Logger logger = LoggerFactory.getLogger(PackageDatabase.class);
 
-    private final Path sourcesFile;
     private final Path databaseFile;
     private final Path installDir;
     private final Gson gson;
     private final List<Package> database;
 
-    PackageDatabase(Path sourcesFile, Path databaseFile, Path installDir) {
-        this.sourcesFile = sourcesFile;
+    PackageDatabase(Path databaseFile, Path installDir) {
         this.databaseFile = databaseFile;
         this.installDir = installDir;
-        gson = new Gson();
+        gson = new GsonBuilder()
+                .registerTypeAdapter(DatabaseRepository.class, new DatabaseRepositoryDeserializer())
+                .create();
         database = loadDatabase();
         markInstalled();
     }
 
     /**
-     * Fetches details of all packages for each repository specified
-     * in {@link #sourcesFile}.
+     * Fetches details of all packages for each repository specified in {@code sourcesFile}.
+     *
+     * @param sourcesFile
      */
-    void sync() {
+    void sync(Path sourcesFile) {
         try (BufferedReader reader = new BufferedReader(
                 new InputStreamReader(Files.newInputStream(sourcesFile))
         )) {
             final List<Package> newDatabase = new LinkedList<>();
-            for (Repository source : gson.fromJson(reader, Repository[].class)) {
-                logger.trace("Fetching package list from: {}", source.url);
+            for (DatabaseRepository source : gson.fromJson(reader, DatabaseRepository[].class)) {
```

Comment:
Please add a `TODO` for parsing the list of repositories beforehand.

Reply:
done